pdns/dnsdistdist/dnsdist-configuration-yaml.cc

```diff
@@ -1159,13 +1160,60 @@ bool loadConfigurationFromFile(const std::string& fileName, [[maybe_unused]] boo
   catch (const std::exception& exp) {
     errlog("Error while processing YAML configuration from file %s: %s", fileName, exp.what());
   }
-  s_registeredTypesMap.lock()->clear();
   return false;
 #else
   (void)fileName;
   throw std::runtime_error("Unsupported YAML configuration");
 #endif /* HAVE_YAML_CONFIGURATION */
 }
+
+void addLuaBindingsForYAMLObjects([[maybe_unused]] LuaContext& luaCtx)
+{
+#if defined(HAVE_YAML_CONFIGURATION)
+  using ReturnValue = boost::optional<boost::variant<std::shared_ptr<DNSDistPacketCache>, std::shared_ptr<DNSRule>, std::shared_ptr<DNSAction>, std::shared_ptr<DNSResponseAction>, std::shared_ptr<NetmaskGroup>, std::shared_ptr<KeyValueStore>, std::shared_ptr<KeyValueLookupKey>, std::shared_ptr<RemoteLoggerInterface>, std::shared_ptr<ServerPolicy>, std::shared_ptr<XSKMap>>>;
+
+  luaCtx.writeFunction("getObjectFromYAMLConfiguration", [](const std::string& name) {
```

Comment:
Small nit: if you add an explicit return type you can do away with `object` and return `boost::none`. The other return values need an explicit conversion (as they already have).